auth_matrix/admin_views.py

```diff
@@ -1,15 +1,27 @@
 from django import views
-from django.contrib.admin.views.decorators import staff_member_required
 from django.contrib.auth import get_user_model
+from django.contrib.auth.mixins import PermissionRequiredMixin, UserPassesTestMixin
 from django.contrib.auth.models import Group, Permission
-from django.utils.decorators import method_decorator
 
 User = get_user_model()
 
-@method_decorator(staff_member_required, name="dispatch")
-class AuthorizationMatrixView(views.generic.base.TemplateView):
 
-    template_name = "admin/authorization/authorization_matrix.html"
+class UserIsStaffMixin(UserPassesTestMixin):
+    def test_func(self):
+        return self.request.user.is_staff
+
+
+def get_auth_matrix_permission():
+    if User._meta.app_label == "auth":
+        return "auth.view_user"
+    return "accounts.view_user"
```

Comment:
the pattern is `<app_label>.view_<model_name>`, so you can build this permission name exactly without guessing:

```py
opts = User._meta
permission = f"{opts.app_label}.view_{opts.model_name}"
```